### PR TITLE
CPDTP-342 Skip mock login for unknown email instead of raising

### PR DESCRIPTION
**Pull request by timabell**
### Context

Resolves an exception seen on sandbox environment when entering in an unknown email address with a domain that is in the `TEST_DOMAINS` list into `/users/sign_in`.

https://sentry.io/organizations/dfe-bat/issues/2509940522/

### Changes proposed in this pull request

Skip mock login for unknown email instead of raising

### Guidance to review

:ship:

### CODE REVIEW

**Review by plisovin — APPROVED** [review]

**Comment by timabell** [comment]
Test coverage on it's way....

**Comment by timabell** [comment]
Excuse the force-push, it adds test coverage to the commit

File: spec/requests/users/sessions_spec.rb [hunk 1]
```diff
@@ -67,29 +67,59 @@ def randomize_case(string)
...
+    let(:environment) { :sandbox }
+
+    before do
+      allow(Rails).to receive(:env).and_return ActiveSupport::EnvironmentInquirer.new(environment.to_s)
+    end
+
+    context "unknown whitelisted email" do
+      let(:test_email) { "not-in-the-database@example.com" }
 
       it "redirects to the dashboard" do
```
Comment (ethanmills) [inline spec/requests/users/sessions_spec.rb:109]: That's not the correct description

File: spec/requests/users/sessions_spec.rb [hunk 2]
```diff
@@ -67,29 +67,59 @@ def randomize_case(string)
...
-    context "using a production environment" do
-      let(:environment) { :production }
+    context "email domain not in the whitelist" do
+      let(:test_email) { "admin@some.other.example.com" }
 
-      it "renders the login_email_sent template" do
-        post "/users/sign_in", params: { user: { email: test_email } }
-        expect(response).to render_template(:login_email_sent)
+      context "using a non-production enviromment" do
+        it "redirects to the dashboard" do
```
Comment (ethanmills) [inline spec/requests/users/sessions_spec.rb:119]: Neither is that

**Review by ethanmills — APPROVED** [review]
Couple of misnamed tests, happy with functionality

File: spec/requests/users/sessions_spec.rb (hunk 1 above)
Comment (timabell) [reply spec/requests/users/sessions_spec.rb]: Oh, good spot, thanks! Done

File: spec/requests/users/sessions_spec.rb (hunk 2 above)
Comment (timabell) [reply spec/requests/users/sessions_spec.rb]: Thanks!! Done

**Pull request merged**